hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java

```diff
@@ -111,12 +127,51 @@ public interface PreHandleContext {
     @NonNull
     PreHandleContext requireKey(@NonNull final Key key);
 
+    /**
+     * Adds the given key to optional non-payer keys.
+     * If the key is the same as the payer key, or if the key has already been added, then the call
+     * is a no-op. The key must not be null.
+     *
+     * @param key key to be added
+     * @return {@code this} object
+     * @throws NullPointerException if the key is null
+     */
+    @NonNull
+    PreHandleContext optionalKey(@NonNull final Key key);
+
+    /**
+     * Adds the given set of keys to optional non-payer keys.
+     * If any key is the same as the payer key, or if any key has already been added, then the call
```

Comment:
```suggestion
     * If any key is invalid, is the same as the payer key, or if any key has already been added, then the call
```

Reply:
This method does not check if the key is valid.  Checking if a key is valid requires extensive work and is only done when signatures are verified.

Reply:
You are correct.  I misread which method this was.
Will fix.